### PR TITLE
Request ID

### PR DESCRIPTION
**Pull request by gabriel**
Implements RequestID.

Also remove InstallID from options since its part of update response and config.

### CODE REVIEW

**Comment by gabriel** [comment]
@keybase/updater-hackers @maxtaco

File: updater.go [hunk 1]
```diff
@@ -36,8 +36,8 @@ type Context interface {
 	BeforeApply(update Update) error
 	AfterApply(update Update) error
 	Restart() error
-	ReportError(err error, options UpdateOptions)
-	ReportAction(action UpdateAction, options UpdateOptions)
+	ReportError(err error, update *Update, options UpdateOptions)
+	ReportAction(action UpdateAction, update *Update, options UpdateOptions)
```
Comment (jzila) [inline updater.go]: Is this change to the exported interface only to enable test validation?

**Comment by jzila** [comment]
Just one pretty big question.

File: updater.go (hunk 1 above)
Comment (gabriel) [inline updater.go]: Nope, it's to give the reporting access to the update, which includes the RequestID returned by the find update request.

**Comment by maxtaco** [comment]
I'm late to the game, but i didn't see how the `requestID` comes back from the `update.json` call and then is threaded through all of the subsequent calls.  Also, I didn't see how `requestID` changes every time through the loop, while `InstallID` stays fixed.

**Comment by gabriel** [comment]
request_id gets passed to reporting stuff via update object... new requestId gets returned by update.json request in each loop I assume?

install_id same thing, except it doesn't change... the installId returned from update.json is either a new one or echoing back the one passed in?

**Comment by maxtaco** [comment]
this makes sense, thanks!

**Pull request merged**